src/System.Net.Primitives/tests/FunctionalTests/IPAddressParsing.cs

```diff
@@ -28,14 +81,118 @@ public void ParseIPv4_WithPort_Failure()
             Assert.Throws<FormatException>(() => { IPAddress.Parse("192.168.0.1:80"); });
         }
 
+        [Fact]
+        public void ParseIPv4_Empty_Failure()
+        {
+            Assert.Throws<FormatException>(() => { IPAddress.Parse(""); });
+        }
+
+        [Theory]
+        [InlineData(" ")]
+        [InlineData(" 127.0.0.1")]
+        public void ParseIPv4_Whitespace_Failure(string address)
+        {
+            Assert.Throws<FormatException>(() => { IPAddress.Parse(address); });
```

Comment:
Nit: the braces and semicolon aren't actually necessary, e.g. it could be:
```C#
Assert.Throws<FormatException>(() => IPAddress.Parse(address));
```
but what you have is of course also ok.